### ISSUE TITLE
Topic level policy support in pulsarctl

### ISSUE THREAD

**Issue by codelipenghui** [streamnative/pulsarctl#246]
The topic level policy is planned to release 2.7.0 and there are some PRs already merged into the Pulsar master branch. The process of the topic level policy support in Pulsar can be found at https://github.com/apache/pulsar/issues/2688. It's better to support topic level policy in pulsarctl.

**Comment by limingnihao** [comment]
I will fix MessageTTL [Get, Set, Remove].

**Comment by limingnihao** [comment]
I will add Max Producer [Get, Set, Remove], and Max Consumer [Get, Set, Remove].

**Comment by limingnihao** [comment]
I will add MaxUnackMessagesPerConsumer [Get, Set, Remove].

**Comment by limingnihao** [comment]
I will add MaxUnackMessagesPerConsumer [Get, Set, Remove].

**Comment by limingnihao** [comment]
I will add MaxUnackMessagesPerSubscription [Get, Set, Remove]

**Comment by limingnihao** [comment]
I will add Persistence Policy [Get, Set, Remove]

**Comment by limingnihao** [comment]
I will add Delayed Delivery Policies [Get, Set, Remove]

**Comment by limingnihao** [comment]
I will add Retention Policy [Get, Set, Remove]

**Comment by limingnihao** [comment]
I will add Message Dispatch Rate  [Get, Set, Remove]

**Comment by limingnihao** [comment]
I will add Deduplication [Get, Set, Remove]

**Comment by nodece** [comment]
I will add Retention Policy [Get, Set, Remove]

**Comment by nodece** [comment]
I will add Backlog Quota [Get, Set, Remove]

**Comment by flowchartsman** [comment]
I added Publish Rate [Get, Set, Remove], even though it is not on the list :)

**Comment by nodece** [comment]
I will add Compaction Threshold [Get, Set, Remove].

**Comment by flowchartsman** [comment]
Is there some context on the deprecation of items 10 and 14? I had been working on a PR for topic-level offload, and I don't see any notes on deprecation in apache/pulsar#2688

**Comment by nodece** [comment]
@flowchartsman I updated description on this issue, you can found the deprecated commands on https://github.com/apache/pulsar/blob/master/pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdTopics.java#L215

**Comment by flowchartsman** [comment]
I am confused, now 10 is marked as done and 14 is deprecated?

**Comment by flowchartsman** [comment]
Inactive topic policy is listed as deprecated in `CmdTopics.java`, but I do not see a reference to offload policy. Unless someone has completed this work already, the link on **10** above is only to the manual offload command not to offload policy.  I didn't see anyone claim it, so I had started the work to support namespace/topic offload policy in my spare time, and I don't see any commits that reference it. Is this work complete? Should I abandon my partial implementation?

**Comment by nodece** [comment]
Hi @flowchartsman, 10 and 14 should be deprecated, I added deprecated commands link. apology :)

**Comment by flowchartsman** [comment]
I am not sure this is correct. On further inspection, it would appear that **neither offload policy nor inactive topic policy** are deprecated. The deprecated commands appear to be specified by comments above newline-separated blocks of commands. Note that there are three such comments, corresponding to three such blocks of commands. Each command in the block is then repeated explicitly in `initDeprecatedCommands()`, and appear to be limited to:

- enable-duplication ([notation](https://github.com/apache/pulsar/blob/85d4c237d0e2caa450eb63e6ae3156f77e1c389d/pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdTopics.java#L134-L137), [deprecation](https://github.com/apache/pulsar/blob/master/pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdTopics.java#L236-L238))
- get-max-unacked-messages-on* ([notation](https://github.com/apache/pulsar/blob/master/pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdTopics.java#L173-L179), [deprecation](https://github.com/apache/pulsar/blob/master/pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdTopics.java#L240-L246))
- maxproducers ([notation](https://github.com/apache/pulsar/blob/master/pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdTopics.java#L194-L197), [deprecation](https://github.com/apache/pulsar/blob/master/pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdTopics.java#L248-L250))

**Comment by nodece** [comment]
You're right. We should to implement the all commands in https://github.com/apache/pulsar/blob/master/pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdTopics.java, and skip the deprecated commands.

**Comment by nodece** [comment]
Hello @codelipenghui and @zymap, I compared `pulsar-admin topics` and `pulsarctl` and sorted out the commands that need to be added to `pulsarctl`, if you are free, please help me to confirm, thanks.

**Comment by nodece** [comment]
I will implement the `Inactive topic policy [Get, Set, Remove]` feature.
